pages/article/[articleLink].js

```diff
@@ -17,14 +18,17 @@ const Article = () => {
 
   const [article, setArticle] = useState();
   const [isDetailsOpen, setIsDetailsOpen] = useState(false);
+  const [user, setUser] = useState();
 
-  const openDetailsHandler = () => {
+  const openDetails = () => {
     setIsDetailsOpen(!isDetailsOpen);
   };
 
   useEffect(() => {
     const handleFetching = async () => {
-      setArticle(await getArticleDetail(articleLink));
+      const data = await getArticleDetail(articleLink);
+      setArticle(data);
+      setUser(await getUserDetail(data.author));
```

Comment:
```suggestion
      const articleDetail = await getArticleDetail(articleLink);
      setArticle(articleDetail);
      
      setUser(await getUserDetail(articleDetail.author));
```

The first two lines fetch and save articleDetail. Then another block of code comes, that fetches and saves the user.

Reply:
Also, doing the fetching first and then the setting of state might save a rerender

```suggestion
      const articleDetail = await getArticleDetail(articleLink);
      const userDetail = await getUserDetail(articleDetail.author)
      
      setArticle(articleDetail);
      setUser(userDetail);
```